prediction_market_agent/agents/microchain_agent/nft_treasury_game/deploy_nft_treasury_game.py

```diff
@@ -71,6 +92,43 @@ def load(self) -> None:
 
         super().load()
 
+    def before_iteration_callback(self) -> None:
+        if self.agent.history and GameRoundEnd.GAME_ROUND_END_OUTPUT in str(
+            self.agent.history[-1]
+        ):
+            # Just sleep for a very long time if the last thing the agent did was being done with this game.
+            # That way he won't be doing anything until the game is reset.
+            logger.info("Agent is done with the game, sleeping.")
+            time.sleep(31_536_000)
+
+    def after_iteration_callback(self) -> None:
+        system_prompt = self.agent.history[0]
+
+        if (
+            not self.game_finished_detected
+            and get_nft_game_status() == NFTGameStatus.finished
+        ):
+            # Switch to more capable (but a lot more expensive) model so that the reflections are worth it.
+            self.agent.llm.generator.model = SupportedModel.gpt_4o.value
+            self.agent.history = [
+                system_prompt,  # Keep the system prompt in the new history.
+                # Hack-in the reasoning in a way that agent thinks it's from himself -- otherwise he could ignore it.
+                {
+                    "role": "assistant",
+                    "content": f"""{Reasoning.__name__}(reasoning='The game is finished. Now the plan is:
```

Comment:
I tried to put this into bootstrap, into the system prompt itself right away, to send it as a message, but nothing worked well.

What seems tow work best is to replace the agent's history with just this reasoning.